internal/backend/remote-state/azure/api_client.go

```diff
@@ -47,54 +47,71 @@ func buildClient(ctx context.Context, config BackendConfig) (*Client, error) {
 		storageAccountName: config.StorageAccountName,
 	}
 
-	// if we have an Access Key - we don't need the other clients
-	if config.AccessKey != "" {
+	var armAuthRequired bool
+	switch {
+	case config.AccessKey != "":
 		client.accessKey = config.AccessKey
-		return &client, nil
-	}
-
-	// likewise with a SAS token
-	if config.SasToken != "" {
+	case config.SasToken != "":
 		sasToken := config.SasToken
 		if strings.TrimSpace(sasToken) == "" {
 			return nil, fmt.Errorf("sasToken cannot be empty")
 		}
 		client.sasToken = strings.TrimPrefix(sasToken, "?")
-
-		return &client, nil
-	}
-
-	if config.UseAzureADAuthentication {
+	case config.UseAzureADAuthentication:
 		var err error
 		client.azureAdStorageAuth, err = auth.NewAuthorizerFromCredentials(ctx, *config.AuthConfig, config.AuthConfig.Environment.Storage)
 		if err != nil {
 			return nil, fmt.Errorf("unable to build authorizer for Storage API: %+v", err)
 		}
+	default:
+		// ARM authN is required only when no auth method is specified, which falls back to listing the access key via ARM API.
```

Comment:
Can you expand what `authN` is? I believe it's ARM Authentication but I just want it to be clear in the comments for when we come back to this

Reply:
Done. AuthN: Authentication, AuthZ: Authorization.